velox/functions/prestosql/types/TimestampWithTimeZoneType.cpp

```diff
@@ -53,15 +77,43 @@ void castFromTimestamp(
   });
 }
 
+void castFromString(
+    const SimpleVector<StringView>& inputVector,
+    exec::EvalCtx& context,
+    const SelectivityVector& rows,
+    VectorPtr& result) {
+  auto timestampWithTzVector = result->asFlatVector<int64_t>();
+  auto rawTimestampWithTzValues =
+      timestampWithTzVector->values()->asMutable<int64_t>();
+  timestampWithTzVector->clearNulls(rows);
+
+  context.applyToSelectedNoThrow(rows, [&](auto row) {
+    if (inputVector.isNullAt(row)) {
+      result->setNull(row, true);
+    } else {
+      rawTimestampWithTzValues[row] =
+          util::Converter<TypeKind::BIGINT>::cast(inputVector.valueAt(row));
```

Comment:
actually, I think that was my bad. You can't use the converter struct as that one is only based on physical types (so this would clash with the actual varchar to bigint conversion). Let's keep this as an anonymous function here, like we do for `castToTimestamp()`.

Reply:
also, don't we need to add the same `adjustTimestampToTimezone` here to adjust it based on the legacy timestamp flag in Presto?

Reply:
Ah that makes sense, didn't realize that. Reverted back to the anonymous function implementation.

And I don't think we need to handle the `adjustTimestampToTimezone` flag - as that's only relevant when dealing with `timestamp` types that don't have an explicit timezone associated with them. In this case, the string we are converting is expected to include a timezone so there would be no ambiguity as to the location it represents. @pedroerp